platform/platform.go

```diff
@@ -260,9 +261,12 @@ func NewMachines(c Cluster, userdata *conf.UserData, n int) ([]Machine, error) {
 // It also ensures the remote system is running Container Linux by CoreOS.
 //
 // TODO(mischief): better error messages.
-func CheckMachine(m Machine) error {
+func CheckMachine(ctx context.Context, m Machine) error {
 	// ensure ssh works and the system is ready
 	sshChecker := func() error {
+		if err := ctx.Err(); err != nil {
```

Comment:
This still feels slightly hacky, but I think it's okay.

Ideally, I think `m.SSH` would also take a ctx, and `util.Retry` would understand if the error returned was a context timeout it's not retryable... but plumbing that through seems like it's overkill for the immediate problem being solved here.